api/src/services/translation.service.ts

```diff
@@ -192,9 +194,33 @@ export class TranslationService {
     listing: Listing,
     language: LanguagesEnum,
   ) {
-    return this.prisma.generatedListingTranslations.findFirst({
-      where: { listingId: listing.id, language: language },
+    const res = await this.prisma.generatedListingTranslations.findFirst({
+      where: {
+        listingId: listing.id,
+        language: language,
+      },
+    });
+
+    //determine if listing or associated preferences have changed since translation creation
+    let mostRecentUpdatedAt = listing.updatedAt;
```

Comment:
I think an issue we are going to run into is that when an application is created for a listing, the listing gets an update to one of its fields, and that'll change the updatedAt value as well.

I think this'll create a cycle where every application will cause listings to get re-translated 

I think we can use this and only redo translations if its been at least 1 day difference on line 213

Reply:
Hmmm I tried adding a paper app and applying online and both times, the updatedAt field was not changed. Do you see the opposite when you run this locally?

Reply:
Per offline discussion, I've updated this logic to fetch new translations if the existing translations are a day old or if there has been a change to the preferences associated with the listing since the last translations. Lmk how that looks to you @YazeedLoonat